jlibra-core/src/main/java/dev/jlibra/serialization/Serializer.java

```diff
@@ -58,6 +77,27 @@ public Serializer appendByte(byte i) {
                 .order(LITTLE_ENDIAN).array();
     }
 
+    /**
+     * https://github.com/zcd/libosu/blob/master/src/main/java/com/zerocooldown/libosu/util/Uleb128.java
+     */
+    private static byte[] intToLeb128ByteArray(int i) {
+        List<Byte> bytes = new ArrayList<>();
+        do {
+            byte b = (byte) (i & MASK_DATA);
+            i >>= 7;
+            if (i != 0) {
+                b |= MASK_CONTINUE;
+            }
+            bytes.add(b);
+        } while (i != 0);
+
+        byte[] ret = new byte[bytes.size()];
+        for (int j = 0; j < bytes.size(); j++) {
+            ret[j] = bytes.get(j);
+        }
+        return ret;
+    }
+
     private Serializer append(byte[] b) {
```

Comment:
any possibility easier to comprehend available? This is very hard to understand. At least a documentation what it should do would be useful here.

Reply:
I added some comments here also and a link to the file where the implementation is from. The uleb128 serialization is a way of efficiently storing integers as bytes and libra is now using this for some values.